Geometry_Engine/Query/HasValue.cs

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2024, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Quantities.Attributes;
+using System;
+using System.ComponentModel;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks whether the input number has a value. If the number is non defined (NaN) or infinite, the method returns false, otherwise true.")]
+        [Input("value", "Number whose value must exist.")]
+        [Output("True if the number has a value (is not NaN or infinite), false if it doesn't.")]
+        public static bool HasValue(this double value)
```

Comment:
Does this need to be a public method available to the entirety of the code base plus UIs?

Reply:
Good catch. Initially I thought it would be helpful, but I ended up changing my mind, in fact it doesn't have references and I wanted to delete it, but I forgot. Deleted in df02debb4fa403fa6c3659e2d11fb5072906c91e